### ISSUE TITLE
ShouldBindQuery should be case insensitive

### ISSUE THREAD

**Issue by axetroy** [gin-gonic/gin#1692]
- With issues:

  - Use the search tool before opening a new issue.
  - Please provide source code and commit sha if you found a bug.
  - Review existing issues and provide feedback or react to them.

- go version: 1.11.2
- gin version (or commit ref):1.3.0
- operating system: mac os

## Description

ShouldBindQuery should be case insensitive

and here is the code

```go
type Query struct {
	Limit    int     `json:"limit"`
	Page     int     `json:"page"`
	Sort     string  `json:"sort"`
	Platform *string `json:"platform"`
}
```

```go
	if err = context.ShouldBindQuery(&query); err != nil {
		return
	}
```

Parse this query `?limit=10&page=1` >>> fail
Parse this query `?Limit=10&Page=1` >>> success

## Screenshots

none

**Comment by egag** [comment]
Because Query strings are not json they won't support the json struct tags. Try using the tag `form` so instead of ````json:"limit"```` it would be ````form:"limit"```` this may support lower case query strings.

Ultimately though, please see https://stackoverflow.com/questions/24699643/are-query-string-keys-case-sensitive which references the RFC standard: https://tools.ietf.org/html/rfc3986#page-11

TL;DR; Query strings should be case sensitive.

**Comment by appleboy** [comment]
improve the testing coverage.

```
func TestContextShouldBindWithQuery(t *testing.T) {
	w := httptest.NewRecorder()
	c, _ := CreateTestContext(w)

	c.Request, _ = http.NewRequest("POST", "/?foo=bar&bar=foo&Foo=bar1&Bar=foo1", bytes.NewBufferString("foo=unused"))

	var obj struct {
		Foo  string `form:"foo"`
		Bar  string `form:"bar"`
		Foo1 string `form:"Foo"`
		Bar1 string `form:"Bar"`
	}
	assert.NoError(t, c.ShouldBindQuery(&obj))
	assert.Equal(t, "foo", obj.Bar)
	assert.Equal(t, "bar", obj.Foo)
	assert.Equal(t, "foo1", obj.Bar1)
	assert.Equal(t, "bar1", obj.Foo1)
	assert.Equal(t, 0, w.Body.Len())
}
```
